[CS2103-F10-3] CS2103/T Revision Tool

khiangleon opened the pull request:
@wilfredbtan @jxofficial @neoshh @ShaunNgTX

khiangleon commented:
@wilfredbtan @jxofficial @neoshh @ShaunNgTX

khiangleon commented:
@wilfredbtan @jxofficial @neoshh @ShaunNgTX

ShirleyWangxt commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,20 +267,64 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 [width="59%",cols="22%,<23%,<25%,<30%",options="header",]
 |=======================================================================
 |Priority |As a ... |I want to ... |So that I can...
-|`* * *` |new user |see usage instructions |refer to instructions when I forget how to use the App
```
The user stories are correctly formatted. All three parts are present.

ShirleyWangxt commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,20 +267,64 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |CS2103 student with a lot of things on my mind |mark certain questions that I am unsure of |refer back to the question when I am free.
+
+|`* * *` |CS2103 student |import questions from my peers |study on my own.
```
The benefit does not match the function. It would be better if the benefit is more specific, such as the student does not need to input the questions himself/herself.

ShirleyWangxt commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,20 +267,64 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* * *` |conscientious CS2103 student |export the questions I am unsure of |raise them up during tutorials.
+
+|`* * *` |indecisive student |be recommended questions instead of me having to plan my own study plan |go directly to studying
```
The benefit is a bit unclear. Hard to see how it matches the function.

shuiyao-sg commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,20 +267,64 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|=======================================================================
 
 [appendix]
 == Use Cases
```
Did you forget to update the Use Cases?

ShirleyWangxt commented on docs/DeveloperGuide.adoc:
```diff
@@ -295,20 +267,64 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+
+|`* *` |competitive CS2103 student |at least know where I stand among my cohort |look at who is the next person I can beat.
+
+|`* *` |gamer CS2103/T student |accomplish tasks that give me a sense of achievement, preferably through in application rewards |I feel good.
```
The benefit can be more specific like I can more motivated to accomplish more tasks.

ShirleyWangxt commented on docs/DeveloperGuide.adoc:
```diff
@@ -279,13 +251,13 @@ Refer to the guide <<DevOps#, here>>.
+*Value proposition*: helps student to ace CS2103/T
 
 [appendix]
 == User Stories
```
Some of the user stories might be a little irrelevant to this app.

irene-lly commented on docs/DeveloperGuide.adoc:
```diff
@@ -121,13 +121,13 @@ image::ModelClassDiagram.png[]
-* stores the Address Book data.
-* exposes an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
+* stores a `Question Bank` object that represents the Question Bank.
+* stores the Test Bank data.
```
A markup can be used here i.e. `Test Bank`

Auxinnn commented on docs/DeveloperGuide.adoc:
```diff
@@ -154,85 +154,57 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
-* **Alternative 1 (current choice):** Saves the entire address book.
+* **Alternative 1 (current choice):** Read and parse the command to execute it
 ** Pros: Easy to implement.
 ** Cons: May have performance issues in terms of memory usage.
```
Maybe it is better to come up with an alternative 2 implementation for comparison?

irene-lly commented on docs/DeveloperGuide.adoc:
```diff
@@ -121,13 +121,13 @@ image::ModelClassDiagram.png[]
 
```
Might want to remove the arrows in composition and aggregation. 
i.e.
![2](https://user-images.githubusercontent.com/54141404/67541982-7cb72380-f71d-11e9-9393-bb50bc0081f5.PNG)
to:
![3](https://user-images.githubusercontent.com/54141404/67542103-fb13c580-f71d-11e9-978c-eb7a62d60194.PNG)

irene-lly replied:
If QuestionList implements ReadOnlyQuestionList then ReadOnlyQuestionList should be an abstract class or an interface.
![1](https://user-images.githubusercontent.com/54141404/67541762-bb98a980-f71c-11e9-80ec-f3115c58f866.PNG)

ShirleyWangxt commented on docs/DeveloperGuide.adoc:
```diff
@@ -12,9 +12,9 @@ ifdef::env-github[]
+By: `Group F10-3`      Since: `Aug 2019`      Licence: `MIT`
 
 == Setting up
 
```
For Figure 2 Class Diagram of the Logic Component, it would be better if the classes and arrows and the different sections can be color-coded to make it easier for visualisation.

ShirleyWangxt commented on docs/DeveloperGuide.adoc:
```diff
@@ -98,9 +98,9 @@ image::LogicClassDiagram.png[]
 *API* :
```
For Figure 3. Component interactions for delete 1 command, the arrow seems to overlap with the activation bar. 
<img width="234" alt="Screenshot 2019-10-25 at 11 37 09 AM" src="https://user-images.githubusercontent.com/43085776/67541649-56dd4f00-f71c-11e9-9f75-a851452be826.png">

For the self-invocation part, it might be better if the line in the middle can be covered.
<img width="191" alt="Screenshot 2019-10-25 at 11 40 03 AM" src="https://user-images.githubusercontent.com/43085776/67541756-b3406e80-f71c-11e9-980f-a7a27a7dbc6a.png">

ShirleyWangxt reviewed (commented):
In general, the diagrams have used intuitive symbols. The description given is understandable and sufficiently high-level. Diagrams are not cluttered with too many unnecessary details. The document uses multiple types of diagrams and they are well-integrated into the description. There are some details to be fixed, but the document looks neat, well-formatted, and professional overall.

irene-lly reviewed (commented):
Overall good job, the diagrams are neat and easy to understand, the document is also pretty well-formatted.

xinmingzh commented on docs/DeveloperGuide.adoc:
```diff
@@ -154,85 +154,57 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
-Step 1. The user launches the application for the first time. The `VersionedAddressBook` will be initialized with the initial address book state, and the `currentStatePointer` pointing to that single address book state.
+Step 1. The user types `add mcq q/"string of question" x/option1 x/option2 y/option3 x/option4 cat/[UML] diff/[easy]`, this command adds a easy difficulty mcq question about UML with 4 options and option3 being the correct answer.
 
 image::UndoRedoState0.png[]
```
The diagrams seems not up to date. Does it match your description?

xinmingzh commented on docs/DeveloperGuide.adoc:
```diff
@@ -154,85 +154,57 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
 
 image::UndoSequenceDiagram.png[]
 
 NOTE: The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
```
Not updated.

xinmingzh commented on docs/DeveloperGuide.adoc:
```diff
@@ -154,85 +154,57 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+===== Aspect: How add executes
 
-* **Alternative 1 (current choice):** Saves the entire address book.
+* **Alternative 1 (current choice):** Read and parse the command to execute it
```
Any alternatives to your current choice?

xinmingzh replied:
Is it possible to be more specific when describing the pros and cons?

wilfredbtan committed:
pass checkstyle

jxofficial committed:
Implement custom mode v1.0

jxofficial committed:
Merge pull request #98 from wilfredbtan/master

Merged Shaun, Sihao. Partial merge of Junxian's Timer and CustomMode. Updated UI as well.

ShaunNgTX committed:
Edit helpCommand URL

ShaunNgTX committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into AutoComplete-feature

# Conflicts:
#	docs/UserGuide.adoc
#	src/main/java/seedu/revision/logic/commands/main/StartQuizCommand.java

jxofficial committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-timer

khiangleon committed:
Update history and stats functionalities

ShaunNgTX committed:
Merge pull request #99 from jxofficial/branch-timer

Implement Custom Mode v1.0

jxofficial committed:
Resolve bug in ArcadeMode

ShaunNgTX committed:
Merge pull request #101 from jxofficial/branch-timer

Resolve bug in ArcadeMode

ShaunNgTX committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into AutoComplete-feature

khiangleon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-status

* 'master' of https://github.com/AY1920S1-CS2103-F10-3/main: (30 commits)
  pass checkstyle
  enable ui to change between mcq, saq, truefalse
  minor ui updates
  added SaqAnswersGridPane to initialise an empty pane for Saq
  pass checkstyle
  clean up SampleDataUtil
  set Answer word limit and update SampleDataUtil
  resolved some travis issues
  add Custom Mode
  Edit StartQuizCommandParser to parse predicates
  update UserGuide, add customisable timer, extend Mode for NormalMode
  -
  Refactor CustomMode to factory to accomodate optional values for timer, category and difficulty
  Add CustomMode.java
  added javadoc for answerchecker class
  update saq AnswerChecker to check for numbers in number format and word format
  Shift order of code in startQuizWindow#executeCommand 1. update bar 2. check if next question is next stage, if yes, confirmation message then create new timer and bar if no, reset timer 3. update the question options and question text
  update QuizCommandParser to parse Saq input
  Fix progress bar to show 8/8 Fix popup window displaying when user inputs command Todo: popup when timer reaches 0 for last question of level
  Make timer stop after every command in StartQuizWindow#executeCommand
  ...

# Conflicts:
#	src/main/java/seedu/revision/ui/StartQuizWindow.java

khiangleon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-status

* 'master' of https://github.com/AY1920S1-CS2103-F10-3/main:
  Resolve bug in ArcadeMode
  Implement custom mode v1.0

# Conflicts:
#	src/main/java/seedu/revision/logic/commands/main/CommandResult.java

khiangleon committed:
Troubleshooting stats and history commands

wilfredbtan committed:
update ui

wilfredbtan committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into pull-custom

wilfredbtan committed:
update user guide with quiz mode

wilfredbtan committed:
add autocomplete to user guide

wilfredbtan committed:
add EOF

wilfredbtan committed:
give Window subclasses their own FXML

wilfredbtan committed:
Merge pull request #105 from wilfredbtan/master

update user guide with quiz mode

khiangleon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-status

* 'master' of https://github.com/AY1920S1-CS2103-F10-3/main:
  add EOF
  update user guide with quiz mode
  update ui
  Edit helpCommand URL
  Updated user guide
  Add "start mode/" command for the autocomplete

wilfredbtan committed:
refactor mode, remove need to get list by mode

khiangleon committed:
Adds new line at EOF

wilfredbtan committed:
add Modes enum

wilfredbtan committed:
make mcq/tf buttons unclickable

wilfredbtan committed:
add icons to quiz popups

wilfredbtan committed:
change ui color

wilfredbtan committed:
add level label and touch up ui

khiangleon committed:
Add sample history

wilfredbtan committed:
add AnswerTest, improve QuestionTest, add McqBuilder class

wilfredbtan committed:
add tests for Mcq class

wilfredbtan committed:
pass checkstyle and tests

wilfredbtan committed:
remove duplicate UserGuide

wilfredbtan committed:
update icon

wilfredbtan committed:
update Ui.png

khiangleon committed:
Merge pull request #148 from wilfredbtan/create-tests

Update UI, add tests for Mcq Class.

wilfredbtan committed:
improve duplicate Answerable check by making notion of equality weaker

wilfredbtan committed:
make valid answer check more stringent by ensuring no duplicates

wilfredbtan committed:
fix timer bug for HelpCommand and restarting timer whenever a wrong input is given

wilfredbtan committed:
fix EditCommand by disallowing change of mcq types or multiple correct, restore list to original size when exiting the quiz

wilfredbtan committed:
add second ui image to display quiz mode

wilfredbtan committed:
add tests for EditCommandTest, EditCommandParserTest, QuizCommandParserTest, AddCommandParserTest, McqTest

wilfredbtan committed:
pass checkstyle

ShaunNgTX committed:
Improve on AutoComplete feature

ShaunNgTX committed:
Updated AutoComplete to Pop Up as user is typing

ShaunNgTX committed:
Merge pull request #149 from wilfredbtan/master

improve stringency of checks for Answer, Mcq, Truefalse and EditCommand. Add tests for EditCommand, QuizCommandParser and others.

ShaunNgTX committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into AutoComplete-feature

ShaunNgTX committed:
Resolved bugs for autocomplete showing even in Quiz mode

wilfredbtan committed:
remove unused logs, improve alert function of StartQuizWindow

wilfredbtan committed:
refactor alerts into a separate class

wilfredbtan committed:
refactor Mode class to allow only CustomMode to be built

wilfredbtan committed:
pass checkstyle, annotate defensive programming methods

wilfredbtan committed:
refactor Answerable and Logic component

wilfredbtan committed:
pass checkstyle

wilfredbtan committed:
refactor Logic component

neoshh committed:
update saq to take in multiple correct answers

khiangleon committed:
updateHistory, handleStats, handleHistory

khiangleon committed:
Merge pull request #150 from ShaunNgTX/AutoComplete-feature

Auto complete feature. updated to make it show as user type

khiangleon committed:
Merge pull request #151 from wilfredbtan/master

Refactor AlertDialog and Mode class. Add Defensive programming.

wilfredbtan committed:
add quiz activity diagram, Answerable Class Diagram, quiz sequence diagram, refactor Answerable input command classes

khiangleon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-status

* 'master' of https://github.com/AY1920S1-CS2103-F10-3/main: (27 commits)
  pass checkstyle, annotate defensive programming methods
  refactor Mode class to allow only CustomMode to be built
  refactor alerts into a separate class
  remove unused logs, improve alert function of StartQuizWindow
  Resolved bugs for autocomplete showing even in Quiz mode
  Updated AutoComplete to Pop Up as user is typing
  Improve on AutoComplete feature
  pass checkstyle
  add tests for EditCommandTest, EditCommandParserTest, QuizCommandParserTest, AddCommandParserTest, McqTest
  add second ui image to display quiz mode
  fix EditCommand by disallowing change of mcq types or multiple correct, restore list to original size when exiting the quiz
  fix timer bug for HelpCommand and restarting timer whenever a wrong input is given
  make valid answer check more stringent by ensuring no duplicates
  improve duplicate Answerable check by making notion of equality weaker
  update Ui.png
  update icon
  remove duplicate UserGuide
  pass checkstyle and tests
  add tests for Mcq class
  add AnswerTest, improve QuestionTest, add McqBuilder class
  ...

# Conflicts:
#	src/main/java/seedu/revision/logic/MainLogicManager.java
#	src/main/java/seedu/revision/ui/StartQuizWindow.java
#	src/test/java/seedu/revision/logic/commands/AddCommandTest.java

wilfredbtan committed:
resolve conflicts with upstream master

wilfredbtan committed:
pass checkstyle

wilfredbtan committed:
resolve conflicts with upstream master

khiangleon committed:
Resolve delete duplicate questions issue

neoshh committed:
Merge pull request #152 from wilfredbtan/master

Add Quiz activity diagram, Answerable Class Diagram, Quiz Sequence diagram

neoshh committed:
Merge branch 'master' into branch-saq-fuzzy-matcher

* master:
  pass checkstyle
  resolve conflicts with upstream master
  add quiz activity diagram, Answerable Class Diagram, quiz sequence diagram, refactor Answerable input command classes
  refactor Logic component
  pass checkstyle
  refactor Answerable and Logic component
  pass checkstyle, annotate defensive programming methods
  refactor Mode class to allow only CustomMode to be built
  refactor alerts into a separate class
  remove unused logs, improve alert function of StartQuizWindow
  Resolved bugs for autocomplete showing even in Quiz mode
  Updated AutoComplete to Pop Up as user is typing
  Improve on AutoComplete feature

khiangleon committed:
Add in difficulty scores for statistics

khiangleon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-status

* 'master' of https://github.com/AY1920S1-CS2103-F10-3/main:
  pass checkstyle
  resolve conflicts with upstream master
  add quiz activity diagram, Answerable Class Diagram, quiz sequence diagram, refactor Answerable input command classes
  refactor Logic component
  pass checkstyle
  refactor Answerable and Logic component

# Conflicts:
#	src/main/java/seedu/revision/logic/LogicManager.java
#	src/main/java/seedu/revision/logic/MainLogic.java
#	src/main/java/seedu/revision/logic/parser/main/MainParser.java
#	src/main/java/seedu/revision/ui/StartQuizWindow.java
#	src/test/java/seedu/revision/logic/LogicManagerTest.java

neoshh committed:
update - fix hanging issue due to stanfordcorenlp library processing when answering saq question

neoshh committed:
update fix saq input parser

neoshh committed:
update AnswerChecker for Saq to take in a list of correct answers instead of just one string of correct answer

neoshh committed:
added fuzzywuzzy library to gradle

neoshh committed:
updated saq answer checker to be more accurate

neoshh committed:
updated travis issues

wilfredbtan committed:
update User Guide

wilfredbtan committed:
udpate UG sections to Configuration Mode and Quiz Mode

ShaunNgTX committed:
Added test for Restore and StartQuiz

ShaunNgTX committed:
Merge https://github.com/AY1920S1-CS2103-F10-3/main into Refactor-AddressBook-naming

ShaunNgTX committed:
Refactor AdressBook to RevisionTool

ShaunNgTX committed:
Refactoring part 2

ShaunNgTX committed:
Merge pull request #153 from neoshh/branch-saq-fuzzy-matcher

Branch saq improvements

ShaunNgTX committed:
Merge https://github.com/AY1920S1-CS2103-F10-3/main into Refactor-AddressBook-naming

wilfredbtan committed:
update PPP till documentation

neoshh committed:
update AnswerChecker check method parameters to take in List instead of ArrayList

neoshh committed:
added AnswerCheckerTest

neoshh committed:
update AnswerCheckerTest

neoshh committed:
updated AnswerChecker check method to take ArrayList as parameters

ShaunNgTX committed:
Refactor part 3

ShaunNgTX committed:
added test for McqInputCommand

wilfredbtan committed:
remove AB3 references

wilfredbtan committed:
remove QuizParserManager dependency

ShaunNgTX committed:
Merge pull request #104 from khiangleon/branch-status

Status and history commands (In-progress)

ShaunNgTX committed:
Refactor 4

wilfredbtan committed:
add ParserClassDiagram to DG

ShaunNgTX committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into Refactor-AddressBook-naming

# Conflicts:
#	src/main/java/seedu/revision/MainApp.java
#	src/main/java/seedu/revision/logic/Logic.java
#	src/main/java/seedu/revision/logic/LogicManager.java
#	src/main/java/seedu/revision/model/Model.java
#	src/main/java/seedu/revision/model/ModelManager.java
#	src/main/java/seedu/revision/model/UserPrefs.java
#	src/main/java/seedu/revision/storage/Storage.java
#	src/main/java/seedu/revision/storage/StorageManager.java
#	src/test/java/seedu/revision/logic/LogicManagerTest.java
#	src/test/java/seedu/revision/logic/commands/AddCommandIntegrationTest.java
#	src/test/java/seedu/revision/logic/commands/AddCommandTest.java
#	src/test/java/seedu/revision/logic/commands/ClearCommandTest.java
#	src/test/java/seedu/revision/logic/commands/DeleteCommandTest.java
#	src/test/java/seedu/revision/logic/commands/EditCommandTest.java
#	src/test/java/seedu/revision/logic/commands/FindCommandTest.java
#	src/test/java/seedu/revision/logic/commands/ListCommandTest.java
#	src/test/java/seedu/revision/model/ModelManagerTest.java

ShaunNgTX committed:
Edit for checkstyle

ShaunNgTX committed:
update autocomplete to include history

neoshh committed:
Merge pull request #154 from ShaunNgTX/Refactor-AddressBook-naming

Refactor address book naming

neoshh committed:
update AnswerCheckerTest checkTest

neoshh committed:
Merge branch 'master' into branch-saq-fuzzy-matcher

* master:
  update autocomplete to include history
  Edit for checkstyle
  Refactor 4
  added test for McqInputCommand
  Refactor part 3
  Refactoring part 2
  Refactor AdressBook to RevisionTool
  Added test for Restore and StartQuiz
  Add in difficulty scores for statistics
  Resolve delete duplicate questions issue
  updateHistory, handleStats, handleHistory
  Add sample history
  Adds new line at EOF
  Troubleshooting stats and history commands
  Update history and stats functionalities
  Link up history and stats with storage, model and logic
  Add stats and history command
  Checkstyle
  Add stats command

neoshh committed:
added Saq questions to SampleDataUtil

wilfredbtan committed:
add wilfred PPP, reformat UG and DG

neoshh committed:
fixed issues

neoshh committed:
update build issues

neoshh committed:
update AnswerCheckerTest

neoshh committed:
update saq AnswerChecker check to take in the whole answerable instead to check answer against keywords from the question

neoshh committed:
update AnswerCheckerTest to include assertFalse

neoshh committed:
updated user guide

neoshh committed:
resolved travis issues

neoshh committed:
resolved build issues

wilfredbtan committed:
update tests and pass checkstyle

wilfredbtan committed:
merge with master

wilfredbtan committed:
clean up McqInputCommandTest

ShaunNgTX committed:
added PPP for ShaunNgTX

wilfredbtan committed:
implement builder design

wilfredbtan committed:
improve code quality by implementing factory design pattern for Answerable and remove redundant dependencies

wilfredbtan committed:
fix bug in StartQuizWindow for normal mode to function properly

wilfredbtan committed:
add GUI tests, improve coveralls from 50% to 72%

wilfredbtan committed:
pass checkstyle

jxofficial committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main

neoshh committed:
update exit command for saq

wilfredbtan committed:
update Answerable Class Diagram

ShaunNgTX committed:
testing for autocomplete. incomplete

ShaunNgTX committed:
Remove AutoComplete test temporary

wilfredbtan committed:
add xvfb service to travis.yml

wilfredbtan committed:
update to JUnit 5

wilfredbtan committed:
update travis header

ShaunNgTX committed:
Merge pull request #158 from wilfredbtan/fix-travis

Add GUI tests, update PPP, fix travis

wilfredbtan committed:
pass Travis CI build

jxofficial committed:
Implement TrueFalseInputCommandTest
Resolve bug where t/f input commands not being marked as case insensitive

neoshh committed:
fixed saq exit issues and invalid answers issues

wilfredbtan committed:
add GUI tests to StartQuizWindowTest

wilfredbtan committed:
add apt-get to travis

wilfredbtan committed:
try workarounds, move guitests into package

khiangleon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-status

* 'master' of https://github.com/AY1920S1-CS2103-F10-3/main: (33 commits)
  update travis header
  update to JUnit 5
  add xvfb service to travis.yml
  update Answerable Class Diagram
  pass checkstyle
  add GUI tests, improve coveralls from 50% to 72%
  fix bug in StartQuizWindow for normal mode to function properly
  improve code quality by implementing factory design pattern for Answerable and remove redundant dependencies
  implement builder design
  clean up McqInputCommandTest
  update tests and pass checkstyle
  add wilfred PPP, reformat UG and DG
  update autocomplete to include history
  Edit for checkstyle
  add ParserClassDiagram to DG
  Refactor 4
  remove QuizParserManager dependency
  remove AB3 references
  added test for McqInputCommand
  Refactor part 3
  ...

neoshh committed:
Conflicts resolved

neoshh committed:
Resolved travis and appveyor build issues

jxofficial committed:
Implement McqInputCommandTest
Implement failing SaqInputCommandTest

jxofficial committed:
-

jxofficial committed:
Merge pull request #155 from neoshh/branch-saq-fuzzy-matcher

Updated Saq AnswerChecker, added Saq AnswerCheckerTest and added Saq questions to SampleDataUtil

ShaunNgTX committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into Refactor-AddressBook-naming

jxofficial committed:
Implement 6 passing tests for TrueFalseTest

wilfredbtan committed:
fix travis by porting AB4 test structure

wilfredbtan committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into pull-master

jxofficial committed:
Merge pull request #161 from wilfredbtan/master

Add GUI tests for StartQuizWindow, fix travis by porting AB4 structure and editing travis.yml

wilfredbtan committed:
Fix formatting bug

jxofficial committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-modes-list-test
Resolve failing tests in SaqInputCommandTest

jxofficial committed:
Implement SaqTest

jxofficial committed:
Implement SaqInputCommandParserTest
Resolve issue #114

wilfredbtan committed:
improve GUI tests

wilfredbtan committed:
update user guide

jxofficial committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main

wilfredbtan committed:
run levelLabelTest in headless

wilfredbtan committed:
update User Guide and About Us page

wilfredbtan committed:
remove AB3 references

wilfredbtan committed:
edit User Guide and PPP

khiangleon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-charts

* 'master' of https://github.com/AY1920S1-CS2103-F10-3/main: (23 commits)
  Fix formatting bug
  fix travis by porting AB4 test structure
  Resolved travis and appveyor build issues
  try workarounds, move guitests into package
  add apt-get to travis
  add GUI tests to StartQuizWindowTest
  fixed saq exit issues and invalid answers issues
  pass Travis CI build
  update exit command for saq
  resolved build issues
  resolved travis issues
  updated user guide
  update AnswerCheckerTest to include assertFalse
  update saq AnswerChecker check to take in the whole answerable instead to check answer against keywords from the question
  update AnswerCheckerTest
  update build issues
  fixed issues
  added Saq questions to SampleDataUtil
  update AnswerCheckerTest checkTest
  updated AnswerChecker check method to take ArrayList as parameters
  ...

ShaunNgTX committed:
update PPP

ShaunNgTX committed:
Merge https://github.com/AY1920S1-CS2103-F10-3/main into Refactor-AddressBook-naming

ShaunNgTX committed:
Merge pull request #163 from wilfredbtan/improve-gui-tests

improve GUI tests and update docs

ShaunNgTX committed:
Updated DG

ShaunNgTX committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into Refactor-AddressBook-naming

wilfredbtan committed:
update delete and quiz sequence diagram

ShaunNgTX committed:
Merge pull request #165 from wilfredbtan/master

update developer guide diagrams

wilfredbtan committed:
update screenshots

wilfredbtan committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main

ShaunNgTX committed:
Update ppp, ug and dg

ShaunNgTX committed:
Merge pull request #167 from wilfredbtan/master

Update screenshots and DG diagrams

ShaunNgTX committed:
Merge https://github.com/AY1920S1-CS2103-F10-3/main into PPP

ShaunNgTX committed:
update dg

ShaunNgTX committed:
update ppp

ShaunNgTX committed:
update ppp

ShaunNgTX committed:
Update ppp

ShaunNgTX committed:
Update AboutUs

ShaunNgTX committed:
update AboutUs

ShaunNgTX committed:
Update UserGuide

ShaunNgTX committed:
Update AutoComplete author

ShaunNgTX committed:
Update author

wilfredbtan committed:
improve code quality but renaming variables reorganising methods

ShaunNgTX committed:
update AutoComplete

ShaunNgTX committed:
minor changes

jxofficial committed:
Merge pull request #168 from ShaunNgTX/PPP

Update PPP, DG and UG

jxofficial committed:
Star question sequence diagram

jxofficial committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-dg-ug

khiangleon committed:
Generate statistics in ui

khiangleon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-charts

* 'master' of https://github.com/AY1920S1-CS2103-F10-3/main: (25 commits)
  minor changes
  update AutoComplete
  Update author
  Update AutoComplete author
  Update UserGuide
  update AboutUs
  Update AboutUs
  Update ppp
  update ppp
  update ppp
  update dg
  Update ppp, ug and dg
  update screenshots
  update delete and quiz sequence diagram
  Updated DG
  update PPP
  edit User Guide and PPP
  remove AB3 references
  update User Guide and About Us page
  run levelLabelTest in headless
  ...

khiangleon committed:
Adjust feedback to user and results display for statscommand and historycommand

khiangleon committed:
Add tests for history and stats

jxofficial committed:
Add UI class diagram
Add EditQuestionAsStarredSequenceDiagram
Add StarQuestionInputSequenceDiagram
Add CodingQuestionActivityDiagram

khiangleon committed:
Update user guide

neoshh committed:
Merge branch 'master' into branch-ppp,-ug,-dg

* master:
  minor changes
  update AutoComplete
  Update author
  Update AutoComplete author
  Update UserGuide
  update AboutUs
  Update AboutUs
  Update ppp
  update ppp
  update ppp
  update dg
  Update ppp, ug and dg
  Updated DG
  update PPP
  Remove AutoComplete test temporary
  testing for autocomplete. incomplete
  added PPP for ShaunNgTX

jxofficial committed:
Add Ui overview in DG

wilfredbtan committed:
update UG, DG, add design considerations and implementation details of Configuration and Quiz Mode

wilfredbtan committed:
resolve upstream conflicts

khiangleon committed:
Add PPP

ShaunNgTX committed:
Update DG for restore and autocomplete

khiangleon committed:
Merge pull request #170 from wilfredbtan/design-considerations

Add design considerations to DG, update UG images, update PPP.

wilfredbtan committed:
update sequence diagrams for delete and quiz

neoshh committed:
Updated developer guide with SaqAnswerChecker class diagram

khiangleon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-charts

* 'master' of https://github.com/AY1920S1-CS2103-F10-3/main:
  update UG, DG, add design considerations and implementation details of Configuration and Quiz Mode
  improve code quality but renaming variables reorganising methods

# Conflicts:
#	src/main/java/seedu/revision/ui/StartQuizWindow.java

wilfredbtan committed:
fix find bug where '?' character is considered a word

jxofficial committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-modes-list-test
Pass checkstyle

jxofficial committed:
Merge pull request #171 from wilfredbtan/design-considerations

Fix sequence diagrams and find command

ShaunNgTX committed:
Updated DG appendix

jxofficial committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-modes-list-test

jxofficial committed:
Resolve Wifred's comments on PR#162

neoshh committed:
Added Saq isCorrect sequence diagram to DG

wilfredbtan committed:
Merge pull request #162 from jxofficial/branch-modes-list-test

Saq, Tf Tests and Mcq, Saq, Tf InputCommandTests

ShaunNgTX committed:
Update autocomplete to remove double enter

jxofficial committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-dg-ug

ShaunNgTX committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into PPP

# Conflicts:
#	docs/DeveloperGuide.adoc

neoshh committed:
added Saq AnswerChecker activity diagram to DG

ShaunNgTX committed:
Update PPP

ShaunNgTX committed:
add newline at EOF

neoshh committed:
Updated dev guide

wilfredbtan committed:
update use case, copy UG DG into PPP

wilfredbtan committed:
Merge pull request #172 from ShaunNgTX/PPP

Updates to PPP, DG and UG

wilfredbtan committed:
merge upstream master

neoshh committed:
updated dev guide

ShaunNgTX committed:
Merge pull request #173 from wilfredbtan/update-PPP

Update ppp and add use cases

wilfredbtan committed:
add manual test case

neoshh committed:
updated user guide

neoshh committed:
updated sihao's PPP

neoshh committed:
updated PPP

neoshh committed:
Merge branch 'master' into branch-ppp,-ug,-dg

* master:
  update use case, copy UG DG into PPP
  add newline at EOF
  Update PPP
  Update autocomplete to remove double enter
  Resolve Wifred's comments on PR#162
  Updated DG appendix
  fix find bug where '?' character is considered a word
  update sequence diagrams for delete and quiz
  Update DG for restore and autocomplete
  update UG, DG, add design considerations and implementation details of Configuration and Quiz Mode
  improve code quality but renaming variables reorganising methods
  Implement SaqInputCommandParserTest Resolve issue #114
  Implement SaqTest
  Implement 6 passing tests for TrueFalseTest
  -
  Implement McqInputCommandTest Implement failing SaqInputCommandTest
  Implement TrueFalseInputCommandTest Resolve bug where t/f input commands not being marked as case insensitive

# Conflicts: RESOLVED
#	docs/DeveloperGuide.adoc

jxofficial committed:
Final PPP, DG, UG

neoshh committed:
resolved travis build issues

neoshh committed:
Merge pull request #175 from neoshh/branch-ppp,-ug,-dg

Updated DG, UG, sihao's PPP

ShaunNgTX committed:
Temp Commit

jxofficial committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-dg-ug

ShaunNgTX committed:
Merge pull request #176 from jxofficial/branch-dg-ug

Final ppp dg ug jx

ShaunNgTX committed:
Test for parser

ShaunNgTX committed:
Merge https://github.com/AY1920S1-CS2103-F10-3/main into PPP

khiangleon committed:
Update DG, UG and PPP

neoshh committed:
updated PPP to within 10 pages

ShaunNgTX committed:
Update ppp

neoshh committed:
Merge pull request #178 from neoshh/branch-ppp,-ug,-dg

updated PPP to within 10 pages

ShaunNgTX committed:
Merge pull request #177 from ShaunNgTX/PPP

Ppp

khiangleon committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into branch-charts

* 'master' of https://github.com/AY1920S1-CS2103-F10-3/main: (28 commits)
  resolved travis build issues
  Final PPP, DG, UG
  updated PPP
  updated sihao's PPP
  updated user guide
  updated dev guide
  update use case, copy UG DG into PPP
  Updated dev guide
  add newline at EOF
  Update PPP
  added Saq AnswerChecker activity diagram to DG
  Update autocomplete to remove double enter
  Added Saq isCorrect sequence diagram to DG
  Resolve Wifred's comments on PR#162
  Updated DG appendix
  fix find bug where '?' character is considered a word
  Updated developer guide with SaqAnswerChecker class diagram
  update sequence diagrams for delete and quiz
  Update DG for restore and autocomplete
  Add Ui overview in DG
  ...

# Conflicts:
#	docs/AboutUs.adoc
#	docs/DeveloperGuide.adoc
#	src/main/java/seedu/revision/ui/StartQuizWindow.java

ShaunNgTX committed:
Merge pull request #179 from khiangleon/branch-charts

DG, UG, PPP

wilfredbtan committed:
Merge branch 'master' of https://github.com/AY1920S1-CS2103-F10-3/main into revert-to-manual

wilfredbtan committed:
omit coveralls

ShaunNgTX committed:
Merge pull request #180 from wilfredbtan/revert-to-manual

Revert tests

wilfredbtan committed:
push latest

ShaunNgTX committed:
Merge pull request #181 from wilfredbtan/pull-master

push latest

neoshh committed:
updated about us to link profiles to correct PPP

neoshh committed:
Merge https://github.com/AY1920S1-CS2103-F10-3/main

jxofficial committed:
Merge pull request #184 from neoshh/master

updated about us to link profiles to correct PPP
